rsnapshot-program.pl

```diff
@@ -4910,13 +4891,25 @@ sub gnu_cp_al {
 	}
 
 	# make the system call to GNU cp
-	$result = system($config_vars{'cmd_cp'}, '-al', "$src", "$dest");
+	eval {
+		$pid = open3(undef, undef, $stderr, $config_vars{'cmd_cp'}, '-al', $src, $dest);
+	};
+
+	if ($@) {
+		$errstr = "Failed to execute $config_vars{'cmd_cp'}: $@";
+		$result = 1;
+	} else {
+		while (<$stderr>) {
```

Comment:
`$errstr = do { local $/; <$stderr> };` ?

Reply:
Please decide. My perl tricks knowledge is basically close to zero.